tests/CommonTestUtils/GitModuleTestHelper.cs

```diff
@@ -13,17 +13,18 @@ namespace CommonTestUtils;
 
 public class GitModuleTestHelper : IDisposable
 {
-#if CI_BUILD
     static GitModuleTestHelper()
     {
+#if CI_BUILD
         Console.WriteLine("GitModuleTestHelper: Disabling explicit clean-up for continuous integration test environment");
-    }
 #else
-    static GitModuleTestHelper()
-    {
         Console.WriteLine("GitModuleTestHelper: Will perform clean-up in background tasks");
+#endif
+
+        TestCleanUp.RegisterCleanUpWait(2, WaitForCleanUpCompletion);
```

Comment:
There's a reason this is 2, even though it's the only one here. 🙂 My local experimental changes for #12556, into which I wrote these changes, register a handler that must run before this and thus which has `order: 1`.

Having just written `order: 1`, I find myself wanting this line to explicitly say `order: 2`, otherwise the 2 is just a magic number. I'll do that later. 🙂

Reply:
This will not fully work as intended because tasks start to run on creation. When or whether they are awaited or not, does not matter.

---

👍 for going to add the arg name `order: 2`